test/unit/plugins/rest_spec.js

```diff
@@ -2021,4 +2021,411 @@ describe("The Rest plugin", function () {
+				getInstanceStub.restore();
+			});
+
+			it("shows the instance does not exist", function () {
+				expect(result.statusCode,"status").to.equal(404);
+				expect(getInstanceStub.args[ 0 ][ 0 ].id).to.equal(INVALID_QUERY);
+			});
+		});
+
+		describe("with a valid instance id", function () {
+			var result;
+			var terminateInstancesStub;
+			var newInstance;
+			var responseCode;
+
+			before(function (done) {
+				//revision reflects the document is updated twice when terminateInstances() is successful.
+				terminateInstancesStub = Sinon.stub(awsAdapter, "terminateInstances", function () {
+					return Bluebird.resolve()
+					.then(function () {
+						done();
+					});
+				});
+
+				return instanceAdapter.createInstance()
+				.then(function (instance) {
+					newInstance = instance;
+					var request = new Request("DELETE", "/v1/instances/" + instance.id);
+					return request.inject(server);
+				})
+				.then(function (response) {
+					result = JSON.parse(response.payload);
+					responseCode = response.statusCode;
+					return Bluebird.resolve();
+				});
+			});
+
+			after(function () {
+				terminateInstancesStub.restore();
+			});
+
+			it("returns the instance state with state set to terminating", function () {
+				expect(responseCode).to.equal(200);
+				expect(result.state).to.equal("terminating");
+			});
+
+			it("status is updated to terminated", function () {
+				return instanceAdapter.getInstance({ id : newInstance.id })
+				.then(function (response) {
+					expect(terminateInstancesStub.called).to.be.true;
+					expect(response.id).to.equal(newInstance.id);
+					expect(response.ami).to.equal(newInstance.ami);
+					expect(response.type).to.equal(newInstance.type);
+					expect(response.state).to.equal("terminated");
+					expect(response.uri).to.equal(newInstance.uri);
+					expect(response.revision).to.be.above(newInstance.revision);
+				});
+			});
+		});
+
+		describe("when there is a AWS error ", function () {
+			var result;
+			var terminateInstancesStub;
+			var newInstance;
+			var responseCode;
+
+			before(function () {
+				terminateInstancesStub = Sinon.stub(awsAdapter, "terminateInstances", function () {
+					return Bluebird.reject(new Error("Simulated Failure"));
+				});
+
+				return instanceAdapter.createInstance()
+				.then(function (instance) {
+					newInstance = instance;
+					var request = new Request("DELETE", "/v1/instances/" + instance.id);
+					return request.inject(server);
+				})
+				.then(function (response) {
+					result = JSON.parse(response.payload);
+					responseCode = response.statusCode;
+					return Bluebird.resolve();
+				});
+			});
+
+			after(function () {
+				terminateInstancesStub.restore();
+			});
+
+			it("returns the instance state with state set to terminating", function () {
+				expect(result.state).to.equal("terminating");
+				expect(responseCode).to.equal(200);
+			});
+
+			it("returns the instance with state set to failed", function () {
```

Comment:
These two `it` statements are a bit conflicting, and could be cleared up. I noticed there's the same problem on line 1534 as well, if you wouldn't mind fixing them both. The second `it` should be talking about the database model, not anything returned.